[Display_01]Build DisplayGpsReceiverImplementationTests

EC01EC opened the pull request:
Working on #67

EC01EC committed:
Create DisplayGpsReceiverImplementationTests.java

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Implemented measureGpsSignalStrength Method. Working on displayGpsManagerIsNotNull Method next.

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Fixed method for getting int value from received signal strength

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Style guide fixes?

EC01EC committed:
Merge branch 'main' into build-DisplayGpsReceiverImplementationTests

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Changes to casting int to Integer and String.

Implemented displayGpsManagerIsNotNull method

EC01EC committed:
Merge branch 'build-DisplayGpsReceiverImplementationTests' of https://github.com/andrewdimmer/csi-3370-software-project into build-DisplayGpsReceiverImplementationTests

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Fixed item passed to DisplayGpsManagerImplementation to Satellite object instead of String

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Null Check

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

removed measureGpsSignalStrength method

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

NotNull Check now includes an Exception thrown if signalStrength is null

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Style Guide "fixes"?

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Style guide "fixes"?

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Style guide "fixes"

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Style guide "fixes"

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

More style guide "fixes"

EC01EC committed:
Merge branch 'main' into build-DisplayGpsReceiverImplementationTests

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Attempts to fix build issues. Not resolved

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Major changes to displayGpsManagerIsNotNull:

- moved null check to ../src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java

- displayGpsManagerIsNotNull now checks for throw of IllegalArgumentException

EC01EC committed:
[Display_01] Build DisplayGpsReceiverImplementation (#140)

* [Database] Build DatabaseGisInterface (#108)

* Made a batabse folde for all

* Update DatabaseGisManager.java

* Update DatabaseGisManager.java

* Create DatabaseGisInterfaceTest.java

* Create DatabaseGisManagerClass.java

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisManagerClass.java

* Update DatabaseGisManager.java

* Update DatabaseGisInterfaceTest.java

* Fixed DGI test class a bit

* more fixes

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisInterfaceTest.java

* Generate SVG and PNG Images for PlantUML Diagrams

* Update DatabasePersistentStorageClass.java

* Fixing tests

* Update DatabasePersistentStorageClass.java

* style fixes

* style fixes

* Update .gitignore

* Created GisManagerStub

* Delete .classpath

* Delete .project

* fixes

* more

* changed location of stub

* removed unwanted files

* Update LocationDataPointTests.java

* Update MiddlewareCommLinkManager.java

* Delete .classpath

* Delete .project

* The Production Class

* fixes

* Delete .project

* deinitalized stuff

* fixes

* overload constructor

* working on it

* Update DatabaseGisManagerStub.java

* Update DatabaseGisInterfaceTest.java

* Class to implementation

* Delete .classpath

* Delete .project

* moved stub to test folder

* Update .gitignore

* Delete #PersistentStorageManager.java#

* Delete placeHolder.txt

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisManagerStub.java

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisManagerStub.java

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisInterfaceImplementation.java

* Update DatabaseGisInterfaceImplementation.java

* Update DatabaseGisInterfaceImplementation.java

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisInterfaceImplementation.java

* Update DatabaseGisManagerStub.java

* Update DatabaseGisInterfaceTest.java

* Update DatabaseGisManagerStub.java

* Update DatabaseGisManagerStub.java

* Update DatabaseGisManagerStub.java

* Update DatabaseGisInterfaceImplementation.java

* Update DatabaseGisInterfaceTest.java

* [Database] Streamline DatabaseGisManagerStub Location Getter

The original design was built to test both a getter and a setter in the DatabaseGisInterface, but the getter is actually in the DatabaseCommInterface so the stub was simplified.

* [Database] Changed Mode Setters Back to Void Return

Was chanced from original spec to return a string for testing; adjusted the testing to return the production spec back to as was mapped out in the UML diagrams.

Co-authored-by: robertsonbrinker <robertsonbrinker@users.noreply.github.com>
Co-authored-by: kylepoterek <60444568+kylepoterek@users.noreply.github.com>
Co-authored-by: Andrew Dimmer <contactandrewd@gmail.com>

* Update DisplayGpsReceiverImplementation.java

Changes made to allow usage for test class

* Update DisplayGpsReceiverImplementation.java

Imported IllegalArgumentException

Co-authored-by: Robertson Brinker <60048070+robertsonbrinker@users.noreply.github.com>
Co-authored-by: robertsonbrinker <robertsonbrinker@users.noreply.github.com>
Co-authored-by: kylepoterek <60444568+kylepoterek@users.noreply.github.com>
Co-authored-by: Andrew Dimmer <contactandrewd@gmail.com>

EC01EC committed:
Merge branch 'main' into build-DisplayGpsReceiverImplementationTests

EC01EC committed:
Updates to DisplayGpsReceiverImplementation and ImplementationTests

Updated test class and production class with changes to method return values

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Removed @return javadoc tag

EC01EC committed:
Updates to DisplayGpsReceiver and DisplayGpsReceiverImplementation(Tests)

Fixes to Production code: DisplayGpsReceiver and DisplayGpsReceiverImplementation

Fixes to Test Class: DisplayGpsReceiverImplementationTests

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Updated missing javadoc tag

andrewdimmer commented on src/test/java/edu/oakland/test/Display01/DisplayGpsReceiverImplementation/DisplayGpsReceiverImplementationTests.java:
```diff
@@ -0,0 +1,21 @@
+
+
+
+@DisplayName ("Display GPS Receiver Implementation")
```
For sake of standardization to make it easier to read and search the logs, can we change this to `DisplayGpsReceiverImplementation Unit Tests`? Thank you!
```suggestion
@DisplayName ("DisplayGpsReceiverImplementation Unit Tests")
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java:
```diff
@@ -1,20 +1,35 @@
-    return "";
-
+    if (satelliteSignal == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```
Just to make this exception a little more descriptive, can we change the output to the name of the parameter? Thank you!
```suggestion
      throw new IllegalArgumentException("satelliteSignal cannot be null");
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java:
```diff
@@ -24,9 +39,9 @@ public String measureGpsSignalStrength(Satellite satelliteSignal) {
-    return null;
-
+    if (satelliteSignal == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```
Just to make this exception a little more descriptive, can we change the output to the name of the parameter? Thank you!
```suggestion
      throw new IllegalArgumentException("satelliteSignal cannot be null");
```

andrewdimmer reviewed (changes requested):
Great start so far! The production code looks good just in terms of a quick read though, we just need to make sure we've got test coverage for it. Based on the code that we have here, there are 4 more tests that you'll need to cover the rest of the production code:

- [x] measureGpsSignalStrengthSatelliteSignalNotNull()
- [x] measureGpsSignalStrengthSignalInIsSignalOut()
- [x] measureSignalSatelliteSignalNotNull()
- [x] measureSignalSignalInIsSignalOut()

To complete these tests, you'll create a new class in the `edu.oakland.test.display01` package called `DisplayGisManagerStub` that implements the `DisplayGisManager` Java interface. For each method defined in the `DisplayGisManagerStub` based on the interface, you'll create and return a new `SatelliteSignalCheckRequest` object with the name of the input satellite as the name of the satellite in the aforementioned output object.

From there, you can simply add the methods for the 4 test cases above.

For each of the "Not Null" cases create a new instance of `DisplayGpsReceiverImplementation` with an instance of `DisplayGisManagerStub` passed in via the constructor, then call the appropriate method based on the test name from within the `assertThrows` test as you currently do for `displayGpsManagerIsNotNull()`. 

For each of the "In Is Out" cases create a new instance of `DisplayGpsReceiverImplementation` with an instance of `DisplayGisManagerStub` passed in via the constructor. Then, you'll create a Satellite object with a non-empty name, and call the appropriate method based on the test name from within the `assertEquals` test to verify that `inputSatellite.getSatelliteName()` is equal to `displayGpsReceiver.methodHere(inputSatellite).getSatelliteName()`. 

If you have any issues or questions, feel free to reach out!

EC01EC committed:
Update src/test/java/edu/oakland/test/Display01/DisplayGpsReceiverImplementation/DisplayGpsReceiverImplementationTests.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

andrewdimmer committed:
Update and rename src/test/java/edu/oakland/test/Display01/DisplayGpsReceiverImplementation/DisplayGpsReceiverImplementationTests.java to src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java

EC01EC committed:
Merge branch 'main' into build-DisplayGpsReceiverImplementationTests

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Fixed missing javadoc summary for DisplayGpsReceverImplementation constructor

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Javadoc summary "too long" fixed

casher-ou committed:
[Display_01] Build stub and test classes

Built DisplayGpsManagerStub
Updated DisplayGpsReceiverImplentationTests

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Accepted changes

EC01EC committed:
Update DisplayGpsManagerStub.java and DisplayGpsReceiverImplementationTests.java

DisplayGpsManagerStub
-Added 'implements DisplayGpsManager' to class line
-Fixed return for measureGpsSignalStrength method

DisplayGpsReceiverImplementationTests
-Accepted changes
-Implemented measureSignalSignalInIsSignalOut method

EC01EC committed:
Updated DisplayGpsManagerStub.java and DisplayGpsReceiverImplementationTests.java

Style Guide fixes for DisplayGpsManagerStub and DisplayGpsReceiverImplementationTests

EC01EC committed:
Updated DisplayGpsManagerStub.java and DisplayGpsReceiverImplementationTests.java

Fixed style guide issues for DisplayGpsManagerStub.java and DisplayGpsReceiverImplementationTests.java.

Added javadoc statements to DisplayGpsReceiverImplementationTests.java

EC01EC committed:
Merge branch 'main' into build-DisplayGpsReceiverImplementationTests

EC01EC committed:
Update DisplayGpsReceiverImplementationTests.java

Fixed "not null" test methods for measureGpsSignalStrengthIsNotNull and measureSignalIsNotNull

stevecatherman reviewed (approved):
Great success!

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java:
```diff
@@ -1,20 +1,36 @@
+   */
+  public DisplayGpsReceiverImplementation(DisplayGpsManager displayGpsManager) {
+    if (displayGpsManager == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```
This is actually checking the manager, not the signal strength.
```suggestion
      throw new IllegalArgumentException("displayGpsManager cannot be null");
```

andrewdimmer commented on src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java:
```diff
@@ -0,0 +1,35 @@
+   * @param satelliteSignal passes satellite object to pull name and checktype from
+   */
+  public String receiveGpsSignalStrength(Satellite satelliteSignal) {
+    SatelliteSignalCheckRequest ssrc = 
+        new SatelliteSignalCheckRequest(satelliteSignal.getSatelliteName(), checkType);
+    return ssrc.getSatelliteName();
```
This just needs to return a string, so we can streamline it as follows:
```suggestion
    return satelliteSignal.getSatelliteName();
```

andrewdimmer commented on src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java:
```diff
@@ -0,0 +1,35 @@
+   * @param satelliteSignal passes satellite object to pull name and checktype from
+   */
+  public SatelliteSignalCheckRequest passGpsSignalStrength(Satellite satelliteSignal) {
+    SatelliteSignalCheckRequest ssrc = 
+        new SatelliteSignalCheckRequest(satelliteSignal.getSatelliteName(), checkType);
```
Just a little cleaner formatting:
```suggestion
    SatelliteSignalCheckRequest ssrc = new SatelliteSignalCheckRequest(
        satelliteSignal.getSatelliteName(),
        checkType
    );
```

andrewdimmer commented on src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java:
```diff
@@ -0,0 +1,64 @@
+  @Test
+  @DisplayName ("Gps Signal Strength Signal Is Not Null")
+  public void measureGpsSignalStrengthSatelliteSignalIsNotNull() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

andrewdimmer commented on src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java:
```diff
@@ -0,0 +1,64 @@
+  @Test
+  @DisplayName ("Satellite Signal Is Not Null")
+  public void measureSignalSatelliteSignalIsNotNull() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

andrewdimmer commented on src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java:
```diff
@@ -0,0 +1,64 @@
+  @Test
+  @DisplayName ("Gps Signal Strength In Is Signal Strength Out ")
+  public void measureGpsSignalStrengthSatelliteInIsSignalOut() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

andrewdimmer commented on src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java:
```diff
@@ -0,0 +1,64 @@
+  @Test
+  @DisplayName ("Measure Signal In Is Signal Out")
+  public void measureSignalSignalInIsSignalOut() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

andrewdimmer reviewed (changes requested):
Few minor comments, mostly around formatting, then you're all set to merge!

EC01EC committed:
Update src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Update src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

andrewdimmer reviewed (approved):
Looks good to me, you're all set to merge provided checks pass!
